controller.py

```diff
@@ -46,7 +47,9 @@ def run(self):
         self.show_boot_message()
 
         while True:
-            k = readkey()
+            k = readkeys.getch()
```

Comment:
This still wasn't working for me on MacOS. But `getkey()` does. Does that work on Windows? Escape key should exit program but, for example, arrow keys shouldn't.

Reply:
I get the following error with replacing readkeys.getch() with readkeys.getkey():
![image](https://github.com/user-attachments/assets/f4d04f44-7500-4a0f-82e2-7733ee28d53d)

Reply:
Okay - let's abort this change. Instead we'll only use ASCII characters as our control keys i.e. get rid of handling the ESC key and use something else, like 'Q', to exit the program. This can be done using the existing readchar module.

Reply:
I've just added a commit that uses different readkeys func depending on the OS, it should work now? If not I'll roll back the changes.